lib/ansible/modules/cloud/amazon/ssm_parameter_store.py

```diff
@@ -0,0 +1,262 @@
+    value: "World"
+
+- name: Delete the key
+  ssm_parameter_store:
+    name: "Hello"
+    state: absent
+
+- name: Create or update secure key/value pair with default kms key (aws/ssm)
+  ssm_parameter_store:
+    name: "Hello"
+    description: "This is your first key"
+    string_type: "SecureString"
+    value: "World"
+
+- name: Create or update secure key/value pair with nominated kms key
+  ssm_parameter_store:
+    name: "Hello"
+    description: "This is your first key"
+    string_type: "SecureString"
+    key_id: "alias/demo"
+    value: "World"
+
+- name: Retrieving plain-text secret
+  ssm_parameter_store:
+    name: "Hello"
+    state: show
+  register: result
+
+- name: Retrieving SecureString secret with default kms key (aws/ssm)
+  ssm_parameter_store:
+    name: "Hello"
+    state: show
+  register: result
+
+- name: Retrieving SecureString secret with nominated kms key
+  ssm_parameter_store:
+    name: "Hello"
+    key_id: "alias/demo"
+    state: show
+  register: result
+
+- name: Retrieving secret without decrypted
+  ssm_parameter_store:
+    name: "Hello"
+    decryption: False
+    state: show
+  register: result
+'''
+
+RETURN = '''
+put_parameter:
+    description: Add one or more paramaters to the system.
+    returned: success
+    type: dictionary
+get_parameter:
+    description: Get details of a parameter.
+    returned: success
+    type: dictionary
+    contains:
+        name:
+            description: The name of the parameter.
+            returned: success
+            type: string
+            sample: "Hello"
+        type:
+            description: The type of parameter. Valid values include [ String, StringList, SecureString ]
+            returned: success
+            type: string
+            sample: "String"
+        value:
+            description: The parameter value.
+            returned: success
+            type: string
+            sample: "World"
+delete_parameter:
+    description: Delete a parameter from the system.
+    returned: success
+    type: dictionary
+'''
+
+try:
+    import botocore
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, AnsibleAWSError, ec2_argument_spec, get_aws_connection_info
+
+
+def create_update_parameter(client, module):
+    changed = False
+
+    args = dict(
+        Name=module.params.get('name'),
+        Value=module.params.get('value'),
+        Type=module.params.get('string_type'),
+        Overwrite=module.params.get('overwrite')
+    )
+
+    if module.params.get('description'):
+        args.update(Description=module.params.get('description'))
+
+    if module.params.get('string_type') == 'SecureString':
+        args.update(KeyId=module.params.get('key_id'))
+
+    try:
+        nacl = client.put_parameter(**args)
+        changed = True
+    except botocore.exceptions.ClientError as e:
```

Comment:
Please follow the [AWS exception guidelines](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-2)